### ISSUE TITLE
accidental variable replacements in maven filtering

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon-ide#128]
In devon-ide we are using maven filtering to replace variables for the release-process within our scripts. However as we are using the default delimiters we also replace variables like `${project.version}` accidentally resulting in this script line:
https://github.com/devonfw/devon-ide/blob/5d4f2cd0d540a63339520fce4c2c8873026cfb12/scripts/src/main/resources/scripts/command/mvn#L47
being replaced as following in our (beta) releases:
```
local version_list="$(mvn -q exec:exec -Dexec.executable=echo -Dexec.args='3.0.0-beta5' ${@})"
```

**Comment by hohwille** [comment]
IMHO the safest way to fix this is to change the delimiter for the filtering and replace the variable syntax in those places where the filtering (replacement of variables) is actually intendet and desired.
However, the drawback is that currently we can use the original scripts using symlinks for testing while in development. This will not work anymore if we change the variable delimiter pattern from `${*}` to e.g. `@@*@@` or something similar as bash understands the `${*}` syntax so we can add something like this to our local `devon.properties` during testing:
```
project.version=3.0.0-beta8
```
Therefore the configurator JAR can still be resolved from here:
https://github.com/devonfw/devon-ide/blob/5d4f2cd0d540a63339520fce4c2c8873026cfb12/scripts/src/main/resources/scripts/functions#L623

This will stop working after the fix...

**Comment by hohwille** [comment]
Maybe we can even add a wildcard for the java version instead of the exact number.
Resolving the devon version (`devon -v`) is another place where this has impact but this should also not be a big issue (might impact testing of `devon ide update`).

**Comment by hohwille** [comment]
BTW: Same error here:
https://github.com/devonfw/devon-ide/blob/5d4f2cd0d540a63339520fce4c2c8873026cfb12/scripts/src/main/resources/scripts/command/mvn#L91
Also resolved due to CI-friendly maven to our release version.

**Comment by hohwille** [comment]
And here:
https://github.com/devonfw/devon-ide/blob/5d4f2cd0d540a63339520fce4c2c8873026cfb12/scripts/src/main/resources/scripts/command/mvn#L97

**Comment by hohwille** [comment]
Finally also here:
https://github.com/devonfw/devon-ide/blob/5d4f2cd0d540a63339520fce4c2c8873026cfb12/scripts/src/main/resources/scripts/command/mvn#L133

**Comment by hohwille** [comment]
Fixed with PR #129
